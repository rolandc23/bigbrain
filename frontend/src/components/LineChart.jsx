import React from 'react';
import { Line } from 'react-chartjs-2';

/*
LineChart
Handles the creation of the line chart
in the admin results page
*/
function LineChart ({ chartData, title, yAxisTitle, xAxisTitle }) {
  return (
    <div className="chart-container">
      <h2 style={{ textAlign: 'center' }}>{title}</h2>
      <Line
        data={chartData}
        options={{
          plugins: {
            legend: {
              display: false
            }
          },
          scales: {
            y: {
              title: {
                display: true,
                text: yAxisTitle
              }
            },
            x: {
              title: {
                display: true,
                text: xAxisTitle
              }
            }
          }
        }
        }
      />
    </div>
  );
}

export default LineChart;
