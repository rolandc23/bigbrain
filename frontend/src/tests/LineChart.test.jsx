import { render, screen } from '@testing-library/react';
import React from 'react';
import LineChart from '../components/LineChart';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

// Nullify chart-js Line element
jest.mock('react-chartjs-2', () => ({
  Line: () => null,
}))

describe('LineChart', () => {
  it('Check linechart', () => {
    // Construct dummy data
    const data = {
      labels: 1,
      datasets: [
        {
          label: 'Test',
          data: 1,
        }
      ]
    }
    render(
      <LineChart chartData={data} title={'Test Title'} yAxisTitle={'TestY'} xAxisTitle={'TestX'} />
    );
    expect(screen.getByText(/Test Title/i)).toBeInTheDocument();

    // Create snapshot of LineChart and confirm it remains consistent with previous tests
    const tree = TestRenderer.create(<LineChart chartData={data} title={'Test Title'} yAxisTitle={'TestY'} xAxisTitle={'TestX'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
