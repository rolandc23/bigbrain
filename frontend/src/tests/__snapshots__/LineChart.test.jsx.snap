// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChart Check linechart 1`] = `
<div
  className="chart-container"
>
  <h2
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Test Title
  </h2>
</div>
`;
